File: Bridge/System/Lazy.cs
using Bridge;
using System.Threading;

namespace System
{
    public class Lazy<T>
    {
        public Lazy()
            : this(LazyThreadSafetyMode.ExecutionAndPublication)
        {
        }

        public Lazy(Func<T> valueFactory)
            : this(valueFactory, LazyThreadSafetyMode.ExecutionAndPublication)
        {
        }

        public Lazy(bool isThreadSafe)
            : this(() => Activator.CreateInstance<T>(), isThreadSafe ? LazyThreadSafetyMode.ExecutionAndPublication : LazyThreadSafetyMode.None)
        {
        }

        public Lazy(Func<T> valueFactory, bool isThreadSafe)
            : this(valueFactory, isThreadSafe ? LazyThreadSafetyMode.ExecutionAndPublication : LazyThreadSafetyMode.None)
        {
        }

        public Lazy(LazyThreadSafetyMode mode)
            : this(() => Activator.CreateInstance<T>(), mode)
        {
        }
        
        public Lazy(Func<T> valueFactory, LazyThreadSafetyMode mode)
        {
        }

        public T Value
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        T InitValue()
        {
            throw new NotImplementedException();
        }

        public bool IsValueCreated
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public override string ToString()
        {
            throw new NotImplementedException();
        }
    }
}

File: Bridge/Resources/.generated/system/lazy.js
    Bridge.define("System.Lazy$1", function (T) { return {
        props: {
            Value: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            },
            IsValueCreated: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            }
        },
        ctors: {
            ctor: function () {
                System.Lazy$1(T).$ctor5.call(this, System.Threading.LazyThreadSafetyMode.ExecutionAndPublication);
            },
            $ctor2: function (valueFactory) {
                System.Lazy$1(T).$ctor4.call(this, valueFactory, System.Threading.LazyThreadSafetyMode.ExecutionAndPublication);
            },
            $ctor1: function (isThreadSafe) {
                System.Lazy$1(T).$ctor4.call(this, function () {
                    return Bridge.createInstance(T);
                }, isThreadSafe ? System.Threading.LazyThreadSafetyMode.ExecutionAndPublication : System.Threading.LazyThreadSafetyMode.None);
            },
            $ctor3: function (valueFactory, isThreadSafe) {
                System.Lazy$1(T).$ctor4.call(this, valueFactory, isThreadSafe ? System.Threading.LazyThreadSafetyMode.ExecutionAndPublication : System.Threading.LazyThreadSafetyMode.None);
            },
            $ctor5: function (mode) {
                System.Lazy$1(T).$ctor4.call(this, function () {
                    return Bridge.createInstance(T);
                }, mode);
            },
            $ctor4: function (valueFactory, mode) {
                this.$initialize();
            }
        },
        methods: {
            initValue: function () {
                throw new System.NotImplementedException();
            },
            toString: function () {
                throw new System.NotImplementedException();
            }
        }
    }; });
